### DISCUSSION TITLE
Cant load a local module

### DISCUSSION THREAD

**Discussion by rafaelmagalhaes** [supabase/supabase#15912] [Questions]
I have a project just for edge functions, I want to share some functions and variables. I created a  `_shared` folder outside the `functions` folder so in the root folder of the project

here's my folder layout

![image](https://github.com/supabase/supabase/assets/2240343/386cf786-7ee4-4543-a8c1-6e393c1f7cda)

in my `_shared` folder I have a file called `supabaseAuth.ts`

    supabase/_shared/supabaseAuth.ts
    export const adminAuthClient = supabaseClient.auth.admin;

in my function when I call the ^(adminAuthClient)  from the shared file I get an error

    serving the request with /home/deno/functions/users
    Download https://deno.land/std@0.194.0/http/server.ts
    InvalidWorkerCreation: worker boot error
        at async Function.create (ext:sb_user_workers/user_workers.js:84:15)
        at async Server.<anonymous> (file:///home/deno/main/index.ts:110:24)
        at async Server.#respond (https://deno.land/std@0.182.0/http/server.ts:220:24)
    CPU time used: 7ms
    Failed to load module: "file:///home/deno/_shared/supabaseAuth.ts" - No such file or directory (os error 2)

in my edge function, I'm calling the file like this

    import { supabaseClient } from "../../_shared/supabaseAuth.ts";

**Comment by GaryAustin1** [comment]
I'm not that great with edge functions stuff, but I believe everything has to be in the functions folder to get uploaded to the deno servers.  They won't have access to anything else.  Supabase examples in supabase/supabase show the shared folder under functions... https://github.com/supabase/supabase/tree/master/examples/edge-functions/supabase/functions

**Comment by caseybaggz** [comment]
I'm getting this same error and all I did was upgrade Docker. 😢 I've tried uninstalling/re-installing Docker & SB CLI and it still crashes so I cannot use my local env at all anymore... Would love not to be blocked by this and actually have a fix so I can ship my app. 😭 

```bash
Applying migration 20230813182909_remote_commit.sql...
Seeding data supabase/seed.sql...
supabase_edge_runtime_virtual-dnd container logs:
Download https://deno.land/std@0.182.0/http/server.ts
Download https://deno.land/x/jose@v4.13.1/index.ts
Sending fatal alert BadCertificate
Failed to load module: "https://deno.land/x/jose@v4.13.1/index.ts" - error sending request for url (https://deno.land/x/jose@v4.13.1/index.ts): error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer

Caused by:
    0: error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer
    1: invalid peer certificate contents: invalid peer certificate: UnknownIssuer
CPU time used: 15ms
Error: worker boot error
service not healthy: [supabase_edge_runtime_virtual-dnd]
Try rerunning the command with --debug to troubleshoot the error.
```

I have 0 edge functions and have never created any.

**Reply by silentworks** [reply]
This is unrelated and not the same issue. Please create a new topic on here with your issue.
